Sources/Commands/SwiftPackageTool.swift

```diff
@@ -0,0 +1,295 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Basic
+import Build
+import Get
+import PackageLoading
+import PackageModel
+import Utility
+import Xcodeproj
+
+#if HasCustomVersionString
+import VersionInfo
+#endif
+
+import enum Build.Configuration
+import enum Utility.ColorWrap
+import protocol Build.Toolchain
+
+import func POSIX.chdir
+
+/// Additional conformance for our Options type.
+extension PackageToolOptions: XcodeprojOptions {}
+
+private enum Mode: Argument, Equatable, CustomStringConvertible {
+    case Init(InitMode)
+    case Doctor
+    case ShowDependencies(ShowDependenciesMode)
+    case Fetch
+    case Update
+    case Usage
+    case Version
+    case GenerateXcodeproj(String?)
+    case DumpPackage(String?)
+
+    init?(argument: String, pop: () -> String?) throws {
+        switch argument {
+        case "init", "initialize":
+            self = try .Init(InitMode(pop()))
+        case "doctor":
+            self = .Doctor
+        case "show-dependencies", "-D":
+            self = try .ShowDependencies(ShowDependenciesMode(pop()))
+        case "fetch":
+            self = .Fetch
+        case "update":
+            self = .Update
+        case "help", "usage", "-h":
+            self = .Usage
```

Comment:
Ah yes, I got right of that by mistake.  I'll add it back and update the PR.  Thanks.